src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -3630,6 +3630,405 @@ void assertReleaseProperties(
+
+		// 9. Create task A
+		String extTaskA = "MAIN/SNOMEDCT-TEST/projectA/taskA";
+		branchService.create(extTaskA);
+
+		// 10. Add new top-level Concept to Extension
+		conceptService.create(
+				new Concept()
+						.addDescription(new Description("Vehicle (vehicle)").setTypeId(FSN).setCaseSignificance(ci).setAcceptabilityMap(intPreferred))
+						.addDescription(new Description("Vehicle").setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intPreferred))
+						.addAxiom(new Relationship(ISA, SNOMEDCT_ROOT)),
+				extTaskA
+		);
+
+		// 13. Promote task A
+		review = getMergeReviewInCurrentState(extProject, extTaskA);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertTrue(conflicts.isEmpty());
+		branchMergeService.mergeBranchSync(extTaskA, extProject, Collections.emptySet());
+
+		// 16. In International, inactivate association member
+		// Author decides to inactivate the member as strictly speaking plural does not necessarily mean SAME_AS
+		String associationMemberId = associationMember.getMemberId();
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		associationMember.setActive(false).updateEffectiveTime();
+		memberService.updateMember(intMain, associationMember);
+
+		// 17. Version International
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220331, "20220331");
+
+		// 18. Upgrade Extension
+		codeSystem = codeSystemService.find("SNOMEDCT-TEST");
+		codeSystemUpgradeService.upgrade(codeSystem, 20220331, false);
+
+		// 19. Assert state of inactivated member before rebase
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331); // International has latest version
+
+		associationMember = memberService.findMember(extProject, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220228); // Not yet rebased latest version
+
+		// 20. Rebase Extension project
+		review = getMergeReviewInCurrentState(extMain, extProject);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertTrue(conflicts.isEmpty());
+		reviewService.applyMergeReview(review);
+
+		// 21. Assert state of inactivated member after rebase
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331);
+
+		associationMember = memberService.findMember(extProject, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331);
+	}
+
+	@Test
+	void testRebaseDoesNotResultInExtensionStealingAssociationMemberWhenConflict() throws ServiceException, InterruptedException {
```

Comment:
This test shows the inactive historical association switches module when there is a conflict. To cause the conflict, the Extension changes the modelling via classification.